wasi_snapshot_preview1/fs.go

```diff
@@ -411,7 +411,12 @@ var fdRead = wasm.NewGoFunc(
 			if errors.Is(err, io.EOF) {
 				break
 			} else if err != nil {
+				if n != 0 { // Let callers process the n > 0 bytes returned before considering the error err.
+					break
+				}
 				return ErrnoIo
+			} else if n < len(b) { // Partial read, don't read into the next buffer.
```

Comment:
cool

Reply:
Yes, this was the one that worried me the most.